salt/etc-hosts/hosts.jinja

```diff
@@ -20,7 +20,7 @@
 	{%- else %}
 		{%- set ip = addrlist|first %}
 	{%- endif %}
-{{ ip }} {{ master_id }} {{ master_id }}.{{ pillar['internal_infra_domain'] }} {{ pillar['api']['server']['external_fqdn'] }}
+{{ ip }} {{ master_id }} {{ master_id }}.{{ pillar['internal_infra_domain'] }} {{ salt['pillar.get']('api:server:external_fqdn', '') }}
```

Comment:
The external FQDN is a required parameter, I can't see a circumstance where it wouldn't be set? I'm guessing we have a bug somewhere else that causes this to be unset? This will just mask that bug.

Reply:
Now the external fqdn field is the last step of the setup, what means that we can:

1. Set the dashboard value
2. Accept minions -> those trigger the `etc-hosts` orchestration
3. Orchestration fails because the setup was not yet finished (`api:server:external` fqdn was not yet set because we didn't finish the setup).

So I'd say this is not really a "bug" but a possibility with the current workflow, and I think it's fine.

Reply:
But this should only run if roles have been set, so I'm also confused now by this change...

Reply:
> The external FQDN is a required parameter, I can't see a circumstance where it wouldn't be set? I'm guessing we have a bug somewhere else that causes this to be unset? This will just mask that bug.

Why is the external FQDN _always_ required? I've never set it and it's always worked fine for me.  It will be required if the user accesses the API server at a external name (so that the kubeconfig/certificates contain that name), but I've always used the internal names fine (because I use the API server from an internal network).

Reply:
How did you manage that? It's a required field, whatever is allowing you to proceed without that is where the bug is:

![image](https://user-images.githubusercontent.com/52713/30216691-bf00605c-94ab-11e7-9315-069cbe45003c.png)

Reply:
@kiall running `salt state.orchestrate orch.kubernetes`. You are forcing the use of the UI, when it is not necessary.

Reply:
Ok, let's do something easier: we can set a default value in `pillar/params.sls` so it can be overwritten from Velum but a `salt state.orchestrate` will not fail either...

Reply:
Yea, we should have a default in there as a reference - as we should for every pillar value we read. That said, you're still deploying CaaSP "wrong" if you don't set the pillar values :)

Reply:
The fact that we added an additional step at the end is a bad UX while leaving the initial screen with the proxy. Yes, this is unrelated to this change, but I needed to say it somewhere. :)